### PR TITLE
oidc-agent: init at 5.1.0

### PR DESCRIPTION
**Pull request by xinyangli**
## Description of changes

`oidc-agent` is a set of tools to manage OpenID Connect tokens and make them easily usable from the command line.

This package is based on https://github.com/indigo-dc/oidc-agent/issues/574. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/oi/oidc-agent/package.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-cOK/rZ/jnyALLuhDM3+qvwwe4Fjkv8diQBkw7NfVo0c=";
+  };
+
+  buildInputs = [
+    pkg-config
+    help2man
+  ];
+
+  nativeBuildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/by-name/oi/oidc-agent/package.nix:31]:
```suggestion
  nativeBuildInputs = [
    pkg-config
    help2man
  ];

  buildInputs = [
```

File: pkgs/by-name/oi/oidc-agent/package.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    libsecret
+    qrencode
+    libsodium
+  ];
+
+  enableParallelBuilding = true;
+
+  makeFlags = [ "PREFIX=$(out)" "BIN_PATH=$(out)" "LIB_PATH=$(out)/lib" ];
+
+  installTargets = "install_bin install_lib install_conf";
```
Comment (SuperSandro2000) [inline pkgs/by-name/oi/oidc-agent/package.nix:44]:
```suggestion
  installTargets = [ "install_bin" "install_lib" "install_conf" ];
```

File: pkgs/by-name/oi/oidc-agent/package.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  postFixup = ''
+    # Override with patched binary to be used by help2man
+    cp -r $out/bin/* bin
+    make install_man PREFIX=$out
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = with lib; {
+    description = "oidc-agent for managing OpenID Connect tokens on the command line";
```
Comment (SuperSandro2000) [inline pkgs/by-name/oi/oidc-agent/package.nix:55]:
```suggestion
    description = "Manage OpenID Connect tokens on the command line";
```

**Comment by xinyangli** [comment]
@SuperSandro2000 resolved and rebased. Thanks for reviewing!

**Pull request merged**